### PR TITLE
Fix/data pragma

### PR DESCRIPTION
**Pull request by lm-sousa**
Fixes issues with `#pragma clava data` that were introduced by the shift to the new APIs.

### CODE REVIEW

File: ClavaWeaver/src/pt/up/fe/specs/clava/weaver/abstracts/ACxxWeaverJoinPoint.java [hunk 1]
```diff
@@ -1034,55 +1032,39 @@ public APragma[] getPragmasArrayImpl() {
...
-            var baseFolder = tu == null ? null
+            File baseFolder = tu == null ? null
                     : tu.getFolderpath().map(folderpath -> new File(folderpath)).orElse(null);
 
             StringSplitter splitter = new StringSplitter(dataPragma.getContent());
-            boolean isDataDirective = splitter.parseTry(StringSplitterRules::string)
-                    .filter(string -> string.toLowerCase().equals(dataKeyword))
+            splitter.parseTry(StringSplitterRules::string)
+                    .filter(string -> string.toLowerCase().equals(ClavaData.KEYWORD_DATA))
                     .isPresent();
```
Comment (joaobispo) [inline ClavaWeaver/src/pt/up/fe/specs/clava/weaver/abstracts/ACxxWeaverJoinPoint.java:1059]: This code is doing nothing

File: ClavaWeaver/resources/clava/test/weaver/PragmaData.lara [hunk 2]
```diff
@@ -30,19 +30,25 @@ aspectdef PragmaData
...
+	$loopWithUpdatedPragma.setData(obj);
 	println("Loop with updated pragma: " + $loopWithUpdatedPragma.pragmas.map(p => p.code));
 
 	// Information persists after rebuilds
 	var $loopBeforeRebuild = Query.search("function", "rebuild").first();
 	println("Is parallel before set and rebuild: " + $loopBeforeRebuild.data.isParallel);
-	$loopBeforeRebuild.data.isParallel = true;
+	obj = $loopBeforeRebuild.data;
+	obj.isParallel = true;
+	$loopBeforeRebuild.setData(obj);
```
Comment (joaobispo) [inline ClavaWeaver/resources/clava/test/weaver/PragmaData.lara:52]: This is a breaking change, add an issue to solve this by using a proxy object that communicates the changes to data in Java.

File: ClavaWeaver/src/pt/up/fe/specs/clava/weaver/abstracts/ACxxWeaverJoinPoint.java [hunk 3]
```diff
@@ -1092,64 +1074,44 @@ public Object getDataImpl() {
...
     public void dataClearImpl() {
-        // Get engine
-        var jsEngine = getWeaverEngine().getScriptEngine();
-
-        var data = getDataImpl();
-        var dataClear = jsEngine.get(data, "_clear");
-
-        // Call _clear()
-        jsEngine.call(dataClear);
+        // TODO: Remove pragma entirely
```
Comment (joaobispo) [inline ClavaWeaver/src/pt/up/fe/specs/clava/weaver/abstracts/ACxxWeaverJoinPoint.java:1113]: Check this TODO

**Review by joaobispo — CHANGES_REQUESTED** [review]

File: ClavaWeaver/src/pt/up/fe/specs/clava/weaver/abstracts/ACxxWeaverJoinPoint.java (hunk 1 above)
Comment (lm-sousa) [reply ClavaWeaver/src/pt/up/fe/specs/clava/weaver/abstracts/ACxxWeaverJoinPoint.java:1059]: Fixed in lastest commit

**Review by joaobispo — APPROVED** [review]

**Pull request merged**